agent/agentssh/agentsshrsa/key.go

```diff
@@ -0,0 +1,85 @@
+package agentsshrsa
+
+import (
+	"crypto/rsa"
+	"math/big"
+	"math/rand"
+)
+
+// DeterministicPrivateKey generates an RSA private key deterministically based on the provided seed.
+// This function uses a deterministic random source to generate the primes p and q, ensuring that the
+// same seed will always produce the same private key. The generated key is 2048 bits in size.
+func DeterministicPrivateKey(seed int64) *rsa.PrivateKey {
+	// Since the standard lib purposefully does not generate
+	// deterministic rsa keys, we need to do it ourselves.
+
+	// Create deterministic random source
+	// nolint: gosec
+	deterministicRand := rand.New(rand.NewSource(seed))
```

Comment:
Would it make more sense to use `math/rand/v2.ChaCha8`? My only reason for suggesting this is that it claims to be "cryptographically strong" https://pkg.go.dev/math/rand/v2#NewChaCha8

At this point it's more of a nice-to-have, as the underlying transport layer also does encryption.